pkg/sources/sources.go

```diff
@@ -42,6 +44,57 @@ type Source interface {
 	GetProgress() *Progress
 }
 
+// Config defines the optional configuration for a source.
```

Comment:
This is an interesting approach that I hadn't thought of. It's nice that we have only one struct for everything, but my concern is it's not clear which options are used for which scan types.

Or do all of the scans support all of these options?

Reply:
I wasn't sure on the req on the tix tbh. I just went with this approach because it made things a bit simpler. Also, I figured doing it this way we can leverage the zero value for the fields in the struct. Then we just configure the struct for each scan independently.

I contemplated a variadic fxn as well, but went with this easier approach. I can make separate structs for each scan, if that's what was needed.

Reply:
Update to use functional args. I think this makes it a bit better?

Reply:
Ooh interesting. And sorry the ticket was underspecified :/

I also hadn't considered using functional args like that. I've seen that pattern used in other libraries except the library would have public functions that returned a private type so they had more control over the options.

I like that it minimizes the risk of null pointers, though technically it's still possible to pass one. We're the only ones using this code so I think it's an acceptable amount of risk though.

> I can make separate structs for each scan, if that's what was needed.

The intention of the ticket was to replace a long list of arguments with something to make the code easier to maintain, update, and read. Imo your solution ticks those boxes.